gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/EnvironmentCommandHandler.java

```diff
@@ -99,13 +101,17 @@ public Single<EnvironmentReply> handle(EnvironmentCommand command) {
             log.info("Environment [{}] handled with id [{}].", environment.getName(), environment.getId());
             return Single.just(new EnvironmentReply(command.getId(), CommandStatus.SUCCEEDED));
         } catch (Exception e) {
-            log.error(
-                "Error occurred when handling environment [{}] with id [{}].",
-                environmentPayload.getName(),
-                environmentPayload.getId(),
-                e
-            );
+            log.error("Error occurred when handling environment [{}] with id [{}].", environmentPayload.getName(), environmentId, e);
             return Single.just(new EnvironmentReply(command.getId(), CommandStatus.ERROR));
         }
     }
+
+    private String getEnvironmentId(EnvironmentPayload environmentPayload) {
+        try {
+            EnvironmentEntity environment = this.environmentService.findByCockpitId(environmentPayload.getCockpitId());
+            return environment.getId();
+        } catch (Exception ex) {
```

Comment:
```suggestion
        } catch (EnvironmentNotFoundException ex) {
```

I would only use the payload id if the environment is not found if a technical error occurs we could create a duplicate 🤔